explorer/interpreter/resolve_names.cpp

```diff
@@ -259,377 +260,404 @@ auto NameResolver::AddExposedNames(const Declaration& declaration,
-        if (auto resolved = alias->resolved_declaration()) {
-          base = *resolved;
-        } else {
-          break;
-        }
-      }
-      if (const auto* namespace_decl = dyn_cast<NamespaceDeclaration>(base)) {
-        auto ns_it = namespace_scopes_.find(namespace_decl);
-        CARBON_CHECK(ns_it != namespace_scopes_.end())
-            << "name resolved to undeclared namespace";
-        CARBON_ASSIGN_OR_RETURN(
-            const auto value_node,
-            ns_it->second.ResolveHere(scope, access.member_name(),
-                                      access.source_loc(),
-                                      /*allow_undeclared=*/false));
-        access.set_value_node(value_node);
-        return {value_node};
-      }
-      break;
-    }
-    case ExpressionKind::CompoundMemberAccessExpression: {
-      auto& access = cast<CompoundMemberAccessExpression>(expression);
-      CARBON_RETURN_IF_ERROR(ResolveNames(access.object(), enclosing_scope));
-      CARBON_RETURN_IF_ERROR(ResolveNames(access.path(), enclosing_scope));
-      break;
-    }
-    case ExpressionKind::IndexExpression: {
-      auto& index = cast<IndexExpression>(expression);
-      CARBON_RETURN_IF_ERROR(ResolveNames(index.object(), enclosing_scope));
-      CARBON_RETURN_IF_ERROR(ResolveNames(index.offset(), enclosing_scope));
-      break;
-    }
-    case ExpressionKind::OperatorExpression:
-      for (Nonnull<Expression*> operand :
-           cast<OperatorExpression>(expression).arguments()) {
-        CARBON_RETURN_IF_ERROR(ResolveNames(*operand, enclosing_scope));
-      }
-      break;
-    case ExpressionKind::TupleLiteral:
-      for (Nonnull<Expression*> field :
-           cast<TupleLiteral>(expression).fields()) {
-        CARBON_RETURN_IF_ERROR(ResolveNames(*field, enclosing_scope));
-      }
-      break;
-    case ExpressionKind::StructLiteral: {
-      std::set<std::string_view> member_names;
-      for (FieldInitializer& init : cast<StructLiteral>(expression).fields()) {
-        CARBON_RETURN_IF_ERROR(
-            ResolveNames(init.expression(), enclosing_scope));
-        if (!member_names.insert(init.name()).second) {
-          return ProgramError(init.expression().source_loc())
-                 << "Duplicate name `" << init.name() << "` in struct literal";
-        }
-      }
-      break;
-    }
-    case ExpressionKind::StructTypeLiteral: {
-      std::set<std::string_view> member_names;
-      for (FieldInitializer& init :
-           cast<StructTypeLiteral>(expression).fields()) {
-        CARBON_RETURN_IF_ERROR(
-            ResolveNames(init.expression(), enclosing_scope));
-        if (!member_names.insert(init.name()).second) {
-          return ProgramError(init.expression().source_loc())
-                 << "Duplicate name `" << init.name()
-                 << "` in struct type literal";
+  return RunWithStackSpace<ErrorOr<std::optional<ValueNodeView>>>(
+      [&]() -> ErrorOr<std::optional<ValueNodeView>> {
```

Comment:
In the cases where you're inserting a lambda in this PR, are you intending to make any non-whitespace changes inside the lambda body?